cmd/ketch/app_deploy_test.go

```diff
@@ -427,3 +429,197 @@ func Test_appDeploy(t *testing.T) {
 		})
 	}
 }
+
+func Test_canaryAppDeploy(t *testing.T) {
+	// safely patch time.Now for tests
+	patch := monkey.Patch(metav1.Now, func() metav1.Time { return metav1.Date(2020, 12, 11, 20, 34, 58, 651387237, time.UTC) })
```

Comment:
no need to use this library there
how about passing in time.Now to appDeploy? 

```golang
type timeNowFn func () time.Time

func appDeploy(..., timeNow timeNowFn, ...) error {
   ....
   nextSch := metav1.NewTime(timeNow().Add(stepInt))
  ...
}

// in production code
appDeploy(..., time.Now, ....)
// in tests
appDeploy(..., func () time.Time { return .... }, ... )
```

Reply:
nice idea. Thanks!